File: 217. Contains Duplicate.java
class Solution {
    public boolean containsDuplicate(int[] nums) {
        Set<Integer> numRead = new HashSet<>();
        for (int i = 0; i < nums.length; i++) {
            if (numRead.contains(nums[i]))
                return true;
            numRead.add(nums[i]);
        }
        return false;
    }
}

File: 2021/217. Contains Duplicate.cpp
class Solution {
public:
    bool containsDuplicate(vector<int>& nums) {
        unordered_set<int> numSet;
        for (int num : nums) {
            if (numSet.count(num)) {
                return true;
            }
            numSet.insert(num);
        }        
        return false;
    }
};

class Solution {
public:
    bool containsDuplicate(vector<int>& nums) {
        unordered_set<int> nSet(nums.begin(), nums.end());
        return nums.size() > nSet.size();
    }
};

class Solution {
public:
    bool containsDuplicate(vector<int>& nums) {
        unordered_set<int> numSet(nums.begin(), nums.end());
        return numSet.size() < nums.size();
    }
};